bokehjs/src/lib/models/annotations/color_bar.ts

```diff
@@ -91,7 +91,31 @@ export class ColorBarView extends AnnotationView {
   }
 
   protected _set_canvas_image(): void {
-    let {palette} = this.model.color_mapper
+    let {palette} = this.model.color_mapper.clone()
```

Comment:
`HasProps.clone()` is a shallow clone of a model, so this line doesn't achieve anything. Instead, use `palette = [...palette, high_color]` and `palette = [low_color, ...palette]` below.

Reply:
However, we don't permit modifying user provided inputs like this. In fact this code may lead to a palette grow indefinitely over the lifetime of this model. A more appropriate solution would be to have a palette member on the (color bar) view, e.g. `extended_palette`, and use this instead (needs to be updated whenever color mapper changes). Though this may still not be sufficient. We may actually create a more sophisticated color bar layout to support this (based on PR #10779).